packages/crud/src/vaadin-crud.js

```diff
@@ -921,73 +910,97 @@ class Crud extends SlotMixin(ControllerMixin(ElementMixin(ThemableMixin(PolymerE
+   * @param {string | RegExp} exclude
+   * @private
+   */
+  __formPropsChanged(form, theme, include, exclude) {
+    if (form) {
+      form.include = include;
+      form.exclude = exclude;
+      form.setAttribute('theme', theme);
+    }
   }
 
-  /** @private */
-  __onCancelButtonChange(cancel, old) {
-    this.__setupSlottedButton(cancel, old, this.__cancelBound);
+  /**
+   * @param {HTMLElement} saveButton
+   * @param {HTMLElement} oldSaveButton
+   * @private
+   */
+  __onSaveButtonChange(saveButton, oldSaveButton) {
+    this.__setupSlottedButton(saveButton, oldSaveButton, this.__saveBound);
   }
 
-  /** @private */
-  __setupSlottedButton(slottedButton, currentButton, clickListener) {
-    if (currentButton && currentButton.parentElement) {
-      currentButton.parentElement.removeChild(currentButton);
+  /**
+   * @param {HTMLElement | undefined} saveButton
+   * @param {string} i18nLabel
+   * @param {boolean} isDirty
+   * @private
+   */
+  __saveButtonPropsChanged(saveButton, i18nLabel, isDirty) {
+    if (saveButton) {
+      saveButton.toggleAttribute('disabled', this.__isSaveBtnDisabled(isDirty));
+      saveButton.textContent = i18nLabel;
     }
-
-    slottedButton.addEventListener('click', clickListener);
   }
 
-  /** @private */
-  __hostPropsChanged() {
-    this.__propagateHostAttributes();
+  /**
+   * @param {HTMLElement} deleteButton
+   * @param {HTMLElement} oldDeleteButton
+   * @private
+   */
+  __onDeleteButtonChange(deleteButton, oldDeleteButton) {
+    this.__setupSlottedButton(deleteButton, oldDeleteButton, this.__deleteBound);
   }
 
-  /** @private */
-  __propagateHostAttributes() {
-    this.__propagateHostAttributesToButton(this._saveButton, HOST_PROPS.save);
-    this.__propagateHostAttributesToButton(this._cancelButton, HOST_PROPS.cancel);
-    this.__propagateHostAttributesToButton(this._deleteButton, HOST_PROPS.delete);
+  /**
+   * @param {HTMLElement | undefined} deleteButton
+   * @param {string} i18nLabel
+   * @param {boolean} isNew
+   * @private
+   */
+  __deleteButtonPropsChanged(deleteButton, i18nLabel, isNew) {
+    if (deleteButton) {
+      deleteButton.textContent = i18nLabel;
+      deleteButton.toggleAttribute('hidden', isNew);
+    }
   }
 
-  /** @private */
-  __propagateHostAttributesToButton(button, props) {
-    // Ensure the slotted button element is present in the DOM.
-    // This is needed because the observer runs before `ready`.
-    if (button) {
-      props.forEach(({ attr, prop, parseProp }) => {
-        if (prop.indexOf('i18n') >= 0) {
-          button.textContent = this.i18n[prop.split('.')[1]];
-        } else {
-          if (typeof parseProp === 'string') {
-            this._setOrToggleAttribute(attr, this[parseProp](this[prop]), button);
-            return;
-          }
+  /**
+   * @param {HTMLElement} cancelButton
+   * @param {HTMLElement} oldCancelButton
+   * @private
+   */
+  __onCancelButtonChange(cancelButton, oldCancelButton) {
```

Comment:
```suggestion
  __onCancelButtonChanged(cancelButton, oldCancelButton) {
```

Reply:
Done in https://github.com/vaadin/web-components/pull/3676.